packages/postcss-normalize-whitespace/src/index.js

```diff
@@ -42,6 +42,55 @@ export default plugin('postcss-normalize-whitespace', () => {
         node.raws.before = node.raws.before.replace(/\s/g, '');
       }
 
+      if (type === atrule && node.params) {
+        let paramClone = node.params;
+        /**
+         * @atRule () , @atRule ()
+         *
+         * => Output
+         *
+         * @atRule (),@atRule ()
+         */
+        paramClone = paramClone
+          .split(',')
+          .map((_) => _.trim())
+          .join(',');
+
+        /**
+         * @atRule and ( ) => @atRule and( )
+         * @atRule and ( )  , => @atRule and( ),
+         */
+        paramClone = paramClone.replace(/\s+\(/g, '(').replace(/\)\s+/g, ')');
+
+        if (/^\(/gi.test(paramClone)) {
+          /**
+           * @atRule ( ){} => @atRule( ) {}
+           */
+          node.raws.afterName = '';
+        }
+
+        /**
+         * removes the whitespace inside of expression i.e ( )
+         * @media (max-width : 49px) {} => @media (max-width:49px) {}
+         * @media (    max-width : 49px) {} => @media (max-width:49px) {}
+         */
+        paramClone = paramClone.replace(/\(.*?\)/gi, (i) =>
+          i.replace(/ +/g, '')
+        );
+
+        // convert all multiple spaces to one in params
+        paramClone = paramClone.replace(/ +/g, ' ');
+        paramClone = paramClone.trim();
+        node.params = paramClone;
+      }
```

Comment:
Using `regex` is unsafe here, we need parser for this stuff, there are a lot of edge cases when we broken syntax

Reply:
I couldn't find any parser that parser at rules params.